configs/templates/generated_interfaces_for_adapters/list/ListProcessor.go

```diff
@@ -0,0 +1,14 @@
+package list
+
+import "istio.io/mixer/configs/templates/list"
+
+type Instance struct {
+  Blacklist       bool
+  CheckExpression interface{}
+}
+
+type ListProcessor interface {
+  ConfigureList(templateName string, types map[string]*foo_bar_mylistchecker.Type /*typeName to Type mapping*/) error
+  CheckList(templateName string, instances map[string]*Instance /*typeName to Instance (generated from Constructor) mapping*/) (bool, error)
```

Comment:
1. You are right, we do not need the templateName since the method signature already implies that. I added it because I was thinking we might want to discuss the case where we do not have template specific method but instead just have Configure and ProcessReport and ProcessCheck methods.. In that case, the argument will be google.protobuf.Struct for instance. Do you know if it is expensive to convert from google.protobuf.Struct to foo.bar.mymetric.Type, given the mixer code has already constructed foo.bar.mymetric.Type and passed to you as a google.protobuf.Struct (since there is no polymorphism , what I am saying is not possible.)?